src/sentry/static/sentry/app/utils/performance/segmentExplorer/segmentExplorerQuery.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
+import {EventQuery} from 'app/actionCreators/events';
+import {LocationQuery} from 'app/utils/discover/eventView';
+import GenericDiscoverQuery, {
+  DiscoverQueryProps,
+  GenericChildrenProps,
+} from 'app/utils/discover/genericDiscoverQuery';
+import withApi from 'app/utils/withApi';
+
+/**
+ * An individual row in a Segment explorer result
+ */
+export type TableDataRow = {
+  id: string;
+  key: string;
+  [key: string]: string | number | TopValue[];
+};
+
+export type TopValue = {
+  name: string;
+  value: string;
+  aggregate: number;
+  count: number;
+};
+
+/**
+ * A Segment Explorer result including rows and metadata.
+ */
+export type TableData = TableDataRow[];
+
+type ChildrenProps = Omit<GenericChildrenProps<TableData>, 'tableData'> & {
+  tableData: TableData | null;
+};
+
+type QueryProps = DiscoverQueryProps & {
+  tagOrder: string;
+  aggregateColumn: string;
+  children: (props: ChildrenProps) => React.ReactNode;
+};
+
+type FacetQuery = LocationQuery &
+  EventQuery & {
+    order?: string;
+    aggregateColumn?: string;
+  };
+
+export function getRequestFunction(_props: QueryProps) {
+  const {tagOrder, aggregateColumn} = _props;
+  function getTagExplorerRequestPayload(props: DiscoverQueryProps) {
+    const {eventView} = props;
+    const apiPayload: FacetQuery = eventView?.getEventsAPIPayload(props.location);
```

Comment:
Why is `eventView` optional here? If it is indeed optional, the two following lines would error right?